src/detection_win.cpp

```diff
@@ -315,11 +302,42 @@ DWORD WINAPI ListenerThread( LPVOID lpParam ) {
 	return 0;
 }
 
+BOOL GetDeviceDetails(HDEVINFO hDevInfo, SP_DEVINFO_DATA* pspDevInfoData, int index, PSP_DEVICE_INTERFACE_DETAIL_DATA* pspDeviceIntDetail) {
+	SP_DEVICE_INTERFACE_DATA spDevIntData;
+	spDevIntData.cbSize = sizeof(SP_DEVICE_INTERFACE_DATA);
+
+	if (!DllSetupDiEnumDeviceInterfaces(hDevInfo, NULL, &GUID_DEVINTERFACE_USB_DEVICE, index, &spDevIntData)) {
+		return false;
+	}
+
+	DWORD dwSize = 0;
+	if (DllSetupDiGetDeviceInterfaceDetail(hDevInfo, &spDevIntData, NULL, 0, &dwSize, NULL)) {
+		return false;
```

Comment:
Why do we return false if this function succeeded?

Reply:
I suppose this check isn't entirely necessary, but is in response to the API documentation stating that it will return false when called to acquire the buffer size.

https://msdn.microsoft.com/en-us/library/windows/hardware/ff551120(v=vs.85).aspx